lib/lambda.js

```diff
@@ -464,7 +466,66 @@ function isLambdaExecutionEnvironment () {
   return !!process.env.AWS_LAMBDA_FUNCTION_NAME
 }
 
+function findHandlerNameInModules (handlerModule, modules) {
+  for (let instrumentedModules of modules) {
+    // array.flat didn't come around until Node 11
+    if (!Array.isArray(instrumentedModules)) {
+      instrumentedModules = [instrumentedModules]
+    }
+    for (const instrumentedModule of instrumentedModules) {
+      if (handlerModule === instrumentedModule) {
+        return true
+      }
+    }
+  }
+  return false
+}
+
+function getFilePath (taskRoot, handlerModule, pathParts) {
+  let filePath = path.resolve(taskRoot, `${handlerModule}.js`)
+  if (!fs.existsSync(filePath)) {
+    filePath = path.resolve(taskRoot, `${handlerModule}.cjs`)
+  }
+  return filePath
+}
+
+function getLambdaHandlerInfo (env, modules, logger) {
+  if (!isLambdaExecutionEnvironment() || !env._HANDLER || !env.LAMBDA_TASK_ROOT) {
+    return
+  }
+
+  // extract module name and "path" from handler using the same regex as the runtime
+  // from https://github.com/aws/aws-lambda-nodejs-runtime-interface-client/blob/c31c41ffe5f2f03ae9e8589b96f3b005e2bb8a4a/src/utils/UserFunction.ts#L21
+  const functionExpression = /^([^.]*)\.(.*)$/
+  const match = env._HANDLER.match(functionExpression)
+  if (!match || match.length !== 3) {
+    return
+  }
+  const handlerModule = match[1]
+  const handlerFunctionPath = match[2]
+
+  // if there's a name conflict with an already instrumented module, skip the
+  // instrumentation of the lambda handle and log a message.
+  if (findHandlerNameInModules(handlerModule, modules)) {
+    logger.info(
+      'Unable to instrument Lambda handler due to name conflict with %s, please choose a different Lambda handler name',
+      handlerModule
+    )
```

Comment:
1 is done in 7b3d65f6a7b64382cd0c22100e21ff1912e8ec73

Re: 2 
> Is this a limitation we could overcome (later)?

Maybe/Probably? Most things are possible with code, it's just a matter of how long it would take.

The challenge her eis there's a lot of hooking and patching code that uniquely identifies a module by its name.  In order to allow a lambda handler's module name to be the same as an instrumented module's name we'd need to change how the agent's hooking/patching code identifies a module, and that might also mean changes to `require-in-the-middle`.  Non trivial planning, non trivial amount of work, and a high potential to break something.